libraries/aws_s3_bucket_object.rb

```diff
@@ -50,6 +50,10 @@ def exists?
     !@bucket_object.nil?
   end
 
+  def resource_id
+    "#{@bucket_name}_#{@key}"
```

Comment:
Here there are two parameters. As "key is not unique. So we have concatenated with the bucket_name following an "_". This can be considered unique.

Reply:
The idea here is not to form an custom unique ID, but rather map the unique ID of the AWS resource so I don't think this is valid
@clintoncwolfe , please correct me if am wrong!